### DISCUSSION TITLE
How to add a new laguage?

### DISCUSSION THREAD

**Discussion by antosha417** [nvim-treesitter/nvim-treesitter#2251] [Q&A] [answered]
Hi! 
I wrote [tree-sitter-hocon](https://github.com/antosha417/tree-sitter-hocon) grammar and want to add hocon support to nvim-treesitter. 
What should I do to add support for new language?

**Comment by theHamsta** [comment, marked as answer]
Add your parser meta data here: https://github.com/nvim-treesitter/nvim-treesitter/blob/a76e1eb89d57e25cedbc71d6bf8e2d2c23d5d85b/lua/nvim-treesitter/parsers.lua#L841-L848 Then, your parser will be installable. Then, you can drop your queries into the `queries` subfolder. Please have a look onto `CONTRIBUTING.md` because we can slightly different highlight groups compared to the Atom ones in the upstream repos.

README is autogenerated. You don't have to change anything on that.
